openmc/data/product.py

```diff
@@ -0,0 +1,187 @@
+from collections.abc import Iterable
+from io import StringIO
+from numbers import Real
+import sys
+
+import numpy as np
+
+import openmc.checkvalue as cv
+from openmc.mixin import EqualityMixin
+from .angle_energy import AngleEnergy
+from .function import Tabulated1D, Polynomial, Function1D
+
+
+class Product(EqualityMixin):
+    """Secondary particle emitted in a nuclear reaction
+
+    Parameters
+    ----------
+    particle : str, optional
+        What particle the reaction product is. Defaults to 'neutron'.
+
+    Attributes
+    ----------
+    applicability : Iterable of openmc.data.Tabulated1D
+        Probability of sampling a given distribution for this product.
+    decay_rate : float
+        Decay rate in inverse seconds
+    distribution : Iterable of openmc.data.AngleEnergy
+        Distributions of energy and angle of product.
+    emission_mode : {'prompt', 'delayed', 'total'}
+        Indicate whether the particle is emitted immediately or whether it
+        results from the decay of reaction product (e.g., neutron emitted from a
+        delayed neutron precursor). A special value of 'total' is used when the
+        yield represents particles from prompt and delayed sources.
+    particle : str
+        What particle the reaction product is.
+    yield_ : openmc.data.Function1D
+        Yield of secondary particle in the reaction.
+
+    """
+
+    def __init__(self, particle='neutron'):
+        self.particle = particle
```

Comment:
Minor nit: It can be a little easier to read when the order of these match the comment above. Other places in the file, such as the ordering of properties and setters match the comment. Though I'd agree that particle could come first here as it is a constructor parameter.